### PR TITLE
feat(link): default rel prop when linking to external sites

### PR DESCRIPTION
**Pull request by chrishavekost**
Provides `rel` with default value of `noopener noreferrer` when linking to external sites. Based on best practice detailed [here](https://developers.google.com/web/tools/lighthouse/audits/noopener).

This will only apply if `target = "_blank"`, `rel` has no specified value, the URL is `absolute`, `loadApp = false`, and the destination hostname is not the same as our current hostname.

`AvLink` has also been modified so that `getUrl` is only called once, and the `absolute` property of the `href` has been extracted for reuse.

### CODE REVIEW

File: packages/link/Link.js [hunk 1]
```diff
@@ -3,32 +3,54 @@ import PropTypes from 'prop-types';
...
-  >
-    {children}
-  </Tag>
-);
+}) => {
+  const absolute = isAbsoluteUrl(href);
+  const url = getUrl(href, loadApp, absolute);
+
+  // eslint-disable-next-line no-cond-assign
+  if ((target = '_blank') && !rel && absolute) {
```
Comment (GoPro16) [inline packages/link/Link.js]: Just wanted to make sure this is correct here. So the only time the first condition would fail is if `target === null`

**Review by GoPro16 — COMMENTED** [review]
Tests look good, just want to confirm that is the only edge case we need to account for above.

File: packages/link/Link.js (hunk 1 above)
Comment (nylon22) [inline packages/link/Link.js]: https://eslint.org/docs/rules/no-cond-assign has this to say:

> There are valid reasons to use assignment operators in conditional statements. However, it can be difficult to tell whether a specific assignment was intentional.

I think we should be explicitly checking if `target` is equal to `"_blank"`
```
if (target === '_blank' && ...

File: packages/link/Link.js [hunk 2]
```diff
@@ -3,32 +3,54 @@ import PropTypes from 'prop-types';
...
+
 const AvLink = ({
   tag: Tag,
-  href: url,
+  href,
   target,
   children,
   onClick,
   loadApp,
+  rel,
```
Comment (nylon22) [inline packages/link/Link.js]: rather than destructuring `rel` from props, leave it out, and if it's specified it will be spread onto the `Tag` and override whatever we set it to.

**Review by nylon22 — CHANGES_REQUESTED** [review]

**Review by nylon22 — APPROVED** [review]
LGTM
